sessions.go

```diff
@@ -186,7 +184,9 @@ func init() {
 
 // Save saves all sessions used during the current request.
 func Save(r *http.Request, w http.ResponseWriter) error {
-	return GetRegistry(r).Save(w)
+	var reg *Registry
+	reg, r = GetRegistry(r)
```

Comment:
If there was no registry saved in the request's context, we shouldn't create a new, empty one here, only to throw it away after telling it to save nothing. Instead, separate `GetRegistry` into its two cases—one that returns an existing registry if it already exists, and one that wraps it and installs a fresh registry if there wasn't one before. This function should use the former and bail early if there was no existing registry.